File: frontend/types/model/schedule.ts
type Schedule = {
  id: string;
  about?: string;
  emoji: string;
  start_time: string;
  end_time?: string;
  place?: string;
  title: string;
  url?: string;
  user_id: string;
};

const TimeFormat = (schedule: Schedule) => {
  const start_time = new Date(schedule.start_time);
  const start_format = `${start_time.getFullYear()}年${
    start_time.getMonth() + 1
  }月${start_time.getDate()}日${start_time.getHours()}:${
    (start_time.getMinutes() < 10 ? '0' : '') + start_time.getMinutes()
  }`;
  const end_time = new Date(schedule?.end_time);
  if (Number.isNaN(end_time.getDate())) {
    return start_format;
  } else {
    const end_format = `${end_time.getHours()}:${(end_time.getMinutes() < 10 ? '0' : '') + end_time.getMinutes()}`;
    return `${start_format}〜${end_format}`;
  }
};

export type { Schedule };
export default TimeFormat;

File: backend/entity/model/schedule.go
package model

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Schedule struct {
	ID        string    `json:"id"`
	About     string    `json:"about" gorm:"default:NULL"`
	Emoji     string    `json:"emoji" gorm:"not null"`
	StartDate time.Time `json:"start_time" gorm:"not null"`
	EndDate   time.Time `json:"end_time" gorm:"default:NULL"`
	Place     string    `json:"place" gorm:"default:NULL"`
	Title     string    `json:"title" gorm:"not null"`
	URL       string    `json:"url" gorm:"default:NULL"`
	UserID    string    `json:"user_id"`
}

func (s *Schedule) BeforeCreate(tx *gorm.DB) (err error) {
	id, _ := uuid.NewUUID()
	s.ID = id.String()
	return
}
